packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,46 +1,42 @@
 import { getActiveTransaction } from '@sentry/tracing';
 
-import { DataFetchingFunction } from './types';
-
 /**
- * Create a span to track the wrapped function and update transaction name with parameterized route.
+ * Call a data fetcher and trace it. Only traces the function if there is an active transaction on the scope.
  *
- * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
- * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
- * @param context The context object passed by nextjs to the function
- * @returns The result of calling the user's function
+ * We only do the following until we move transaction creation into this function: When called, the wrapped function
+ * will also update the name of the active transaction with a parameterized route provided via the `options` argument.
  */
-export async function wrapperCore<T extends DataFetchingFunction>(options: {
-  origFunction: T['fn'];
-  context: T['context'];
-  route: string;
-  op: string;
-}): Promise<T['result']> {
-  const { origFunction, context, route, op } = options;
+export async function callDataFetcherTraced<F extends (...args: any[]) => Promise<any> | any>(
+  origFunction: F,
+  origFunctionArgs: Parameters<F>,
+  options: {
```

Comment:
what do you think about making this the `SpanContext` instead of options? we would make `route` -> `description`.

Reply:
This is blocked on https://github.com/getsentry/sentry-javascript/pull/5564. I have to see what we're gonna do with op and description there before I make any more changes here.

Honestly, I wouldn't read too much into this PR. We're likely gonna scrap this wrapper anyways because it's too generic to be used across different data fetching methods.